### PR TITLE
VEN-26 | Switch to GraphQL

### PR DESCRIPTION
**Pull request by Nizar-Rahme**

### CODE REVIEW

File: src/components/berths/Berth/SelectedBerth/index.tsx [hunk 1]
```diff
@@ -77,7 +76,7 @@ class SelectedBerth extends Component<Props, State> {
               })}
             >
               <span key={berth.identifier}>
-                {index + 1}. {getLocalizedText(berth.name, intl.locale)}
+                {index + 1}. {berth.name}
```
Comment (midori0507) [inline src/components/berths/Berth/SelectedBerth/index.tsx]: Are you sure about berth name shouldn't be translated here? I bet it doesnt need to, just asking

File: src/index.tsx [hunk 2]
```diff
@@ -30,15 +32,28 @@ Sentry.init({
   dsn: REACT_APP_SENTRY_DSN
 });
 
+const client = new ApolloClient({
```
Comment (midori0507) [inline src/index.tsx]: This config shouldn't be a part of index.tsx
Can you make new file under config maybe? and the URI should come from ENV

**Comment by midori0507** [comment]
All mock node REST should be removed as part of this PR

File: src/index.tsx [hunk 3]
```diff
@@ -30,15 +32,28 @@ Sentry.init({
...
+  uri: 'https://venepaikka-api-gql.test.hel.ninja/graphql/',
```
Comment (igordavydsson) [inline src/index.tsx]: shouldn't this be configurable through an environment variable?

File: src/utils/graphql.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import gql from 'graphql-tag';
+
+export const BOATTYPES_BERTHS_QUERY = gql`
+  {
```
Comment (rBoost) [inline src/utils/graphql.ts]: We should use named queries: `query BoatTypesBerthsQuery {`.

File: src/components/common/BoatsBerthsQuery/index.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+          maximumWidth: number;
+          maximumLength: number;
+          maximumDepth: number;
+        };
+      };
+    }>;
+  };
+}
+
+class BoatsBerthsQuery extends Query<Data> {}
```
Comment (rBoost) [inline src/components/common/BoatsBerthsQuery/index.tsx]: Why do we need this? I would just use `Query` component.

Comment (Nizar-Rahme) [reply src/components/common/BoatsBerthsQuery/index.tsx]: For adding types to the `data` object which Query component attaches to the render prop function.

File: .env.example [hunk 6]
```diff
@@ -5,3 +5,4 @@ REACT_APP_MAX_SELECTED_BERTHS=10
 REACT_APP_PIWIK_URL=https://your.piwik-url.com
 REACT_APP_PIWIK_ID=piwik_id
 REACT_APP_SENTRY_DSN=https://id1@sentry.io/id2
+REACT_APP_GRAPHQL_URI=https://venepaikka-api-gql.test.hel.ninja/graphql/
```
Comment (igordavydsson) [inline .env.example]: weeeellll, we are gonna destroy this instance after the merge. maybe `http://localhost:8000/graphql/`?

Comment (igordavydsson) [reply .env.example]: or something more examplary. not a real endpoint

Comment (Nizar-Rahme) [reply .env.example]: Oh, I see!
In this case, we have to implement a mock server for doing the job.

Comment (igordavydsson) [reply .env.example]: I just meant that I would prefer not to put even our test endpoint here. But I don't know, maybe there is no harm if people know its URL.

File: src/utils/graphql.ts [hunk 7]
```diff
@@ -0,0 +1,59 @@
+import gql from 'graphql-tag';
+
+export const BOATTYPES_BERTHS_QUERY = gql`
```
Comment (rBoost) [inline src/utils/graphql.ts]: Shouldn't this be `BOAT_TYPES_BERTHS_QUERY`?

Comment (midori0507) [reply src/utils/graphql.ts]: @Nizar-Rahme

**Pull request merged**